internal/driver/drv_sysfs.go

```diff
@@ -0,0 +1,147 @@
+// +build sysfs
+
+package driver
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/spf13/cast"
+)
+
+type GPIODev struct {
+	lc   logger.LoggingClient
+	gpio uint8
+}
+
+func NewGPIODev(lc logger.LoggingClient) *GPIODev {
+	return &GPIODev{lc: lc, gpio: 0}
+}
+
+func (dev *GPIODev) SetGPIO(gpio string, value bool) error {
+	if !strings.Contains(gpio, "/") {
+		return errors.New("invalid gpio number")
+	}
+	split := strings.Split(gpio, "/")
+	valid_port, err := cast.ToUint8E(split[1])
+	if err != nil {
+		return err
+	}
+	if err := export(valid_port); err != nil {
+		return err
+	}
+	if err = setDirection(valid_port, "out"); err != nil {
```

Comment:
I think it would also be nice to remember the direction of each pin, and only set it if/when it needs to be changed.